shared/test/test_assets.rb

```diff
@@ -454,4 +446,21 @@ def post_file(assets, channel_id, filename, contents, content_type)
     response = post(assets, channel_id, file)
     [response, tmp_filename]
   end
+
+  def assert_custom_metric(index, metric_type, length_msg = nil)
```

Comment:
It looks like this is hard-coded to FilesApi custom metrics.  Could you pass in the type of the custom metric as a parameter (e.g. "FilesApi") to make this more general and explicit.